### DISCUSSION TITLE
Code reuse between web, mobile and desktop

### DISCUSSION THREAD

**Discussion by jordan-lumley** [jonataslaw/getx#2141] [Q&A]
Hey all, 

Looking for some ideas/input on if some of you are running into some architectural road blocks like us. 

We have a linux desktop app that looks at its own PostgresQL database. When running this as a web project, obviously postgres cannot be run in the browser and talk to a database. Have any of you using GetX or even flutter in general run your projects on two different platforms? And if so, would something just like a n-tier application structure work? With a web API. 

Note: when its running in desktop mode, it will talk to its own localhost database. Code wise with a repo/service layer, any examples how to make local database and web API providers work would be greatly appreciated.

**Comment by gOzaru** [comment]
You should ask this on stackoverflow.com. 
Not here.
